packages/repository/src/repositories/legacy-juggler-bridge.ts

```diff
@@ -210,6 +213,71 @@ export class DefaultCrudRepository<T extends Entity, ID>
     );
   }
 
+  /**
+   * @deprecated
```

Comment:
Let's not add methods that are already deprecated at creation time. Please remove `_createHasAndBelongsToManyRepositoryFactoryFor` from the pull request.